src/encoded/batch_download.py

```diff
@@ -291,6 +328,139 @@ def _get_annotation_metadata(request, search_path, param_list):
         content_disposition='attachment;filename="%s"' % 'metadata.tsv'
     )
 
+def _get_publicationdata_metadata(request):
+    """
+    Generate PublicationData metadata.tsv.
+
+        :param request: Pyramid request
+    """
+    qs = QueryString(request)
+    param_list = qs.group_values_by_key()
+
+    # Get the required "dataset={path}" parameter
+    dataset_path = param_list.get('dataset', [''])[0]
+
+    # Open the metadata.tsv file for writing
+    fout = io.StringIO()
+    writer = csv.writer(fout, delimiter='\t')
+
+    # Build the column-title header row and write it to the file.
+    header = [header for header in _tsv_mapping_publicationdata if header not in _excluded_columns]
+    writer.writerow(header)
+
+    # Load the specified PublicationData object and extract its files to build the rows
+    dataset = request.embed(dataset_path, as_user=True)
+    file_ids = dataset.get('files', [])
+    if file_ids:
+        # For each file, load the file object and use its properties to build each row of the table.
+        for file_id in file_ids:
+            # Load the file object and disqualify those we don't handle.
+            file = request.embed(file_id, as_user=True)
```

Comment:
I feel like we might have to batch these in groups of 500 or something. There are publicationDatasets with 10,000+ files.    So like /search/ with @fields=... *&limit=500?  (500 was just an arbitrary guess)

Reply:
Does request.embed do a big search? I pass it an individual file `@id` here, so I thought it wasn’t too bad. Granted, the file_ids list could be huge so this request.embed could happen apparently 10,000+ times.